### DISCUSSION TITLE
lxqt-panel: Disappear after turning off and on screen

### DISCUSSION THREAD

**Discussion by Ra2-IFV** [lxqt/lxqt#2635] [Q&A]
##### Expected Behavior

panel should appear after I turning on my screen

##### Current Behavior

panel is gone after I turning of the screen

##### Possible Solution

dont know
##### Steps to Reproduce (for bugs)

1. Run `lxqt-panel`
2. Turn off and on screen
3. Panel does not show

##### Context

nah, not critical but quite annoying because I will have to use `lxqt-config-session` to restart `lxqt-panel`

##### System Information

* Distribution & Version: `Arch Linux`
* Kernel: `6.11.5-arch1-1`
* Qt Version: `6.8.0-1`
* liblxqt Version: `2.0.0.r7.g3c8db8e-1`
* Package version: `2.0.1.r56.g194bbabe-1`

Using `labwc` and `sddm`, start lxqt with `startlxqtwayland`
Didn't see related issues.
Log is here, I didn't find any debug options.
```text
$ lxqt-panel
No user preferences available. Attempting auto-detection.

Panel backend: "libwmbackend_wlroots.so" 

Supplied protocol version to QWaylandClientExtensionTemplate is higher than the version of the protocol, using protocol version instead.
StatusNotifierProxy, services: QList()
*TURN OFF SCREEN*
"Failed to get sink information: No such entity"
"Failed to get sink information: No such entity"
"Failed to get sink information: No such entity"
qt.qpa.wayland: Creating a fake screen in order for Qt not to crash
qt.qpa.wayland: Creating a fake screen in order for Qt not to crash
"Failed to get sink information: No such entity"
"Failed to get sink information: No such entity"
*TURN ON SCREEN*
```

**Comment by tsujan** [comment]
It has never happened here, with LXQt+labwc (or kwin_wayland or wayfire).

If lxqt-panel crashed, please attach its backtrace. If there's no crash in lxqt-panel, the cause should be outside LXQt.

How do you turn off the screen?

**Reply by Ra2-IFV** [reply]
No, it's not crashing, just not showing up.
I press the power button on the monitor, that's all.

**Reply by tsujan** [reply]
> No, it's not crashing, just not showing up.

The problem should be related to the compositor, either directly or indirectly. You could know that by checking if it also happens under X11.

**Reply by Ra2-IFV** [reply]
Removing the configuration doen't help...
I'll report if I find anything useful. Thank you very very much, I love LXQt.

**Reply by Ra2-IFV** [reply]
I switched to `labwc` `autostart` to start `lxqt-panel` but it doesn't help, `labwc -d` doesn't output any related things.
```
$ QT_DEBUG_PLUGINS=1 QML_IMPORT_TRACE=1 QT_MESSAGE_PATTERN='[%{time process} %{type}] %{appname} %{category} %{function} - %{message}' lxqt-panel
......
[     7.421 warning] lxqt-panel default unknown - "Failed to get sink information: No such entity"
[     7.421 warning] lxqt-panel default unknown - "Failed to get sink information: No such entity"
[     7.421 warning] lxqt-panel default unknown - "Failed to get sink information: No such entity"
[     7.467 info] lxqt-panel qt.qpa.wayland unknown - Creating a fake screen in order for Qt not to crash
[    12.372 warning] lxqt-panel default unknown - "Failed to get sink information: No such entity"
[    12.372 warning] lxqt-panel default unknown - "Failed to get sink information: No such entity"
[    12.372 warning] lxqt-panel default unknown - "Failed to get sink information: No such entity"
```
`Failed to get sink information: No such entity` comes from the volume widget

**Comment by tsujan** [comment]
Google shows many instances of `qt.qpa.wayland: Creating a fake screen in order for Qt not to crash`.

Converting the report to a discussion....

**Reply by Ra2-IFV** [reply]
not useful

**Comment by themanyone** [comment]
Try enabling power management in xscreensaver-settings. It seems to have fixed issues with panel and icons disappearing randomly, desktop background image corruption after ~10 minutes. The delay between startup and when problem appears was the clue.

It all started after removing laptop battery unknowingly disabled power management during startup.

Fedora 39 + LXQT + EliteBook 8760w + proprietary NVIDIA driver

Another quickfix: Sometimes right-clicking around where the panel should be (sometimes it appears hidden on another monitor. I have three) pops up a menu to "configure panel" Click it and -> reserve space on display. Panel and icons reappear and everything... until screen stops updating the background behind windows again in ~10 minutes. 

The problem appears mostly-FIXED. It happens a lot less frequently, for me now at least, with xscreensaver power management.

- **_The duration is tied to the time to wait before screen saver is set to come on._**

 (Regardless of whether or not there is an actual screen saver enabled). Now what is to blame for this interaction?

**Reply by themanyone** [reply]
After soft "dnf upgrade" to Fedora 41, latest (for old laptop) proprietary kmod-nvidia-6.12.6-200.fc41 driver.
Everything is working great! For me at least.

**Comment by Ra2-IFV** [comment]
I'm using wayland.
I believe it's a compositor problem because I see weird color pixels in the background after that, and I just found a related issue at labwc/labwc/issues/2304

**Reply by tsujan** [reply]
> ... I see weird color pixels in the background after that

If you have a translucent panel, that's a regression in Qt 6.8.

**Reply by Ra2-IFV** [reply]
Sorry they are actually not on the panel,
In Desktop Perferences: background color is black, wallpaper is set
After toggling the output with `wlr-randr` the background area that not covered by the wallpaper is filled with random pixels, and the panel is gone but still running (same with the issue)
`qdbus6 org.kde.StatusNotifierWatcher /global_key_shortcuts/panel/fancymenu/show_hide org.lxqt.global_key_shortcuts.client.activated` can open the menu.
`wlopm` will not cause this.

**Reply by Ra2-IFV** [reply]
by the way, when I'm adjusting the `background opacity` in panel settings, the slider becomes very laggy and unresponsive, it takes about 2~3 secs to change from 100% to 0% (so with the panel), driver is properly installed and hardware acceleration should work.

**Comment by stefonarch** [comment]
I wonder if this issue I see on Niri is related:
https://github.com/lxqt/lxqt-panel/issues/2170

But panel.conf isn't updated on lid close.

**Reply by tsujan** [reply]
I see no relation.

**Reply by stefonarch** [reply]
With some configuration I've no top panel after suspend, that's probably the only common thing here. We have only "monitor 1 monitor 2",  I noticed  that xfce4-panel has "automatic, primary, and the identifiers to choose from:

![image](https://github.com/user-attachments/assets/b101a2a4-c632-48ae-b732-129b9a163c3e)
 `Output "Philips Consumer Electronics Company PHL 243V7 UK02348055187" (HDMI-A-1)`
 
 With this option the panel doesn't move to the other output if its output is disconnected as lxqt-panel does always, allowing having a panel dedicated for an external monitor.

**Reply by tsujan** [reply]
You last comment was the most informative on this page. It suggests that, in such cases, something about the screens changes after resuming from suspension, and that's problematic because resuming shouldn't change anything by definition. In other words, the problem is not in the panel.

If I find some time, I'll look into the code to see whether a workaround is possible.

**Reply by stefonarch** [reply]
My guess is that what I see on Niri could be a smithay issue - the monitors are behaving like if they are  unplugged and not at the same time. I've no idea how monitor are identified across X11 and wayland in lxqt-panel. Nice would be having: 
* what we have now → panel switches to the remaining monitor
* monitor with identifier → is only shown when the monitor is connected
* on all monitors

**Reply by stefonarch** [reply]
Looks I pasted the wrong link, the issue is this
https://github.com/YaLTeR/niri/issues/987#issuecomment-2593783872

**Reply by YaLTeR** [reply]
Hey, to add some context from that issue.

Addressing monitors by index doesn't make much sense on Wayland: there's no "index" that is communicated to the client, and the order of wl_output or xdg_output globals is not defined to mean anything. (It seems that part of the issue is that on niri this order happens to change occasionally.) Addressing by "primary" also doesn't seem to make sense because there's no "primary" flag in either wl_output or xdg_output if I'm not mistaken.

You can pretty much only address by monitor name (usually it's the connector name like HDMI-A-1) and description (the Philips Consumer ... string).

**Reply by stefonarch** [reply]
Yes, but the drawback is that the panel will disappear if the monitor is disconnected, which can be desired (dedicated panel for that monitor) or not (only one panel). Unless we could add some option.

I had a look at all compositors we support with no explicit monitor settings:
* no one does this switch between session start and after suspend. 
* All see the laptop monitor at (0,0) and as  "1" (in lxqt-panel) and put the external one to the right by default
* except sway which puts it on the left and sees the external one as "1" and 
* kwin which asks what to do and if set to "right" it sees also the external as "1", just to be different ;) . 

Assuming so  "1" or "primary" could be the one at (0,0).

**Reply by tsujan** [reply]
> Addressing monitors by index doesn't make much sense on Wayland: there's no "index"

Fixed for both panel and desktop in git. The fixes will be in LXQt 2.2.0.
